src/backend/db/database_manager.py

```diff
@@ -1813,4 +1813,262 @@ def get_paginated_files(self, customer_guid: str, page: int = 1, page_size: int
+                    'status': result.status,
+                    'errors': result.errors,
+                    'error_retry': result.error_retry,
+                    'completed_time': result.completed_time,
+                    'to_be_deleted': result.to_be_deleted,
+                    'delete_request_timestamp': result.delete_request_timestamp,
+                    'delete_status': result.delete_status,
+                    'final_delete_timestamp': result.final_delete_timestamp
+                })
+            else:
+                # Update existing record in common_db
+                update_common_query = """
+                    UPDATE common_db.customer_file_status
+                    SET to_be_deleted = :to_be_deleted,
+                    delete_request_timestamp = :delete_request_timestamp,
+                    delete_status = :delete_status
+                    WHERE customer_guid = :customer_guid AND filename = :filename
+                """
+                session.execute(text(update_common_query), {
+                    'customer_guid': customer_guid,
+                    'filename': filename,
+                    'to_be_deleted': result.to_be_deleted,
+                    'delete_request_timestamp': result.delete_request_timestamp,
+                    'delete_status': result.delete_status
+                })
+
+            session.commit()
+            logger.info(f"Marked file {filename} for deletion for customer {customer_guid}")
+            return True
+
+        except SQLAlchemyError as e:
+            logger.error(f"Error marking file for deletion: {e}")
+            session.rollback()
+            return None
+        finally:
+            session.close()
+
+    def get_deletion_status(self, customer_guid: str, filename: str):
+        """Get current deletion status and retry count"""
+        session = self._session_factory()
+        try:
+            query = """
+            SELECT delete_status, error_retry 
+            FROM common_db.customer_file_status 
+            WHERE customer_guid = :customer_guid AND filename = :filename
+            """
+            result = session.execute(text(query), {
+                'customer_guid': customer_guid,
+                'filename': filename
+            }).fetchone()
+            return result if result else None
+        except SQLAlchemyError as e:
+            logger.error(f"Error getting deletion status: {e}")
+            return None
+        finally:
+            session.close()
+
+    def get_pending_deletions(self, max_threads):
+        """Get files marked for deletion that need processing"""
+        session = self._session_factory()
+        try:
+            num_of_records = max_threads * 4
+            query = """
+            SELECT customer_guid, filename 
+            FROM common_db.customer_file_status 
+            WHERE to_be_deleted = TRUE
+            ORDER BY delete_request_timestamp ASC
```

Comment:
make it to order by current_activity_updated_time

return fileid instead of filename